src/ingestion/kafka-queue.ts

```diff
@@ -80,24 +84,25 @@ export class KafkaQueue implements Queue {
             },
         })
         this.wasConsumerRan = true
+        return promise
```

Comment:
Why not await the promise here as well? I mean, in the end it works the same way (this `queue.start` is always directly awaited if I see correctly?), but just `await`ing here would show intent a bit better. Maybe I'm missing something :)

Reply:
Hm, alright. Also put in all of start() into the promise while at it.

Reply:
Actually also put `eachBatch` in its own `KafkaQueue` method, much cleaner this way.

Reply:
Hm... is this 100% correct though? Could be, but looking at it:
- Is there a chance the callbacks run and the promise resolves before the `wasConsumerRan = true` line, causing it not to run? Maybe Node will still run that code, but I'm not sure.
- Those `on` callbacks. They're not cleared in `stop()`, which is used when reloading the server. Could they linger on and cause issues?

Reply:
* Overall the method should be just fine, just moved that `wasConsumerRan = true` before `consumer.run()`
* Not really an issue, they don't do anything besides resolving/rejecting a promise, which is idempotent. Possibly a tiny tiny memory leak in case of an unexpectedly large number of reconnects? But unlikely to be a problem.

Reply:
the plugin server might have to reload often, so this should be fixed somehow, probably...